mod-source-record-manager-server/src/main/java/org/folio/services/MappingRuleCache.java

```diff
@@ -0,0 +1,65 @@
+package org.folio.services;
+
+import com.github.benmanes.caffeine.cache.AsyncLoadingCache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.dao.MappingRuleDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+@Component
+public class MappingRuleCache {
```

Comment:
Let`s add some docs here for methods for better understanding this mechanism.

Reply:
done